go/worker/storage/committee/node.go

```diff
@@ -629,6 +630,43 @@ func (n *Node) worker() { // nolint: gocyclo
 
 	n.logger.Info("starting committee node")
 
+	nodeCh, nodeSub, err := n.commonNode.Consensus.Registry().WatchNodes(n.ctx)
+	if err != nil {
+		n.logger.Error("worker/storage: failed to watch registry nodess",
+			"err", err,
+		)
+		return
+	}
+
+	// Watch for storage node registration updates for the runtime and update the policy.
+	go func() {
```

Comment:
Should this, together with the subscription above, be moved to its own function and then just spawned here?

Reply:
yeah can do that

Reply:
done (also some other minor changes)

Reply:
Actually thinking a bit more about this, there's some more things that i'd like to fix, i'll move this back to a draft PR for now:
- avoid doing the query that queries ALL nodes while having the crossNode lock, instead the new watcher can maintain an up to date list of all registered storage nodes
https://github.com/oasisprotocol/oasis-core/blob/398e7186ccf243b0d07c8a0a5fb17dac38a5d213/go/worker/storage/committee/node.go#L394-L397
- this new code resulted in some unneeded calls to the `updateExternalServicePolicyLocked` since it's also called for all registered compute worker updates (while before it was just for compute nodes in the committee)
- There is also a race condition i think - it can happen now that the new watcher gets a (compute) committee node update notification before the node descriptor in the snapshot is updated (since we trigger `updateExternalServicePolicyLocked` from the new watcher, while we still query the `group.EpochSnapshot` for descriptors)

Reply:
@kostko  please take a look again, i think this is a more proper solution now